src/include/migraphx/op/dequantizelinear.hpp

```diff
@@ -54,7 +54,7 @@ struct dequantizelinear
         {
             MIGRAPHX_THROW("DEQUANTIZELINEAR: Zero point and input should be the same type.");
         }
-        return {inputs[1].type(), inputs[0].lens(), inputs[0].strides()};
+        return inputs[0].with_lens(inputs[1].type(), inputs[0].lens());
```

Comment:
Wouldn't `inputs[0].with_type(inputs[1].type())` do?

Reply:
I think he needs to set things based on output type as the dequantized output type is based on the scale type used.

Reply:
The suggested simpler code is doing just that. Use scale type. Thanks.

Reply:
> Wouldn't inputs[0].with_type(inputs[1].type()) do?

That only changes the type, it doesnt change the strides to make them packed.

Reply:
Shouldn't the api `with_type` be fixed for this? All that is expected is the `type` to change, nothing else. Thanks.

Reply:
I meant to fix the `with_type()` for its default behavior. Anyways, it is a discussion for some other time. Thanks.